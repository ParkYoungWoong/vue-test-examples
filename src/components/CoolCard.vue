<template>
  <div
    class="vte__cool-card"
    data-test="cool-card-div"
  >
    hello I am a card :)
  </div>
</template>

<script>
export default {
  name: 'CoolCard',
  props: {
    wow: {
      type: String,
      required: false
    }
  },
  data() {
    return {
      buzzword: ''
    };
  },
  computed: {
    someComputedProperty() {
      return `${this.buzzword} is very big`;
    }
  },
  methods: {
    someMethod(arg1, arg2) {
      return arg1 + arg2;
    }
  }
};
</script>

<style>
.vte__cool-card {
  height: 20vh;
  width: 40vw;
  padding: 2rem;
  border-radius: 1rem;
  box-shadow: 0px 1px 1px #000;
}
</style>
