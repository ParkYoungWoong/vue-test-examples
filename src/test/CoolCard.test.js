import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import CoolCard from '../components/CoolCard.vue';
import StoreFactory from './utils/StoreFactory';

const localVue = createLocalVue();
localVue.use(Vuex);

const actions = {
  incrementTotal: jest.fn()
};

describe('CoolCard', () => {
  let store;
  beforeEach(() => {
    store = StoreFactory(
      {
        basket: { currentTotal: 567 }
      },
      actions
    );
  });
  test('should render content correctly', () => {
    const wrapper = mount(CoolCard, { store, localVue });
    expect(wrapper.find('[data-test="cool-card-div"]').text()).toEqual(
      'hello I am a card :)'
    );
  });
  test('someMethod should add numbers together', () => {
    const wrapper = mount(CoolCard, { store, localVue });
    expect(wrapper.vm.someMethod(1, 2)).toEqual(3);
  });
  test('someComputedProperty should evaluate buzzwords correctly', () => {
    const wrapper = mount(CoolCard, {
      store,
      localVue,
      propsData: {
        wow: 'wow'
      }
    });
    wrapper.setData({
      buzzword: 'Big Data'
    });
    expect(wrapper.vm.someComputedProperty).toEqual('Big Data is very big');
  });
  test('card should show correct total', () => {
    const wrapper = mount(CoolCard, { store, localVue });
    expect(wrapper.vm.currentTotal).toEqual(567);
  });
});
