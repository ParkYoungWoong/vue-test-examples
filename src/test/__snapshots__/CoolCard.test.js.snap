// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoolCard should render content correctly 1`] = `
<div
  class="vte__cool-card"
>
  <p
    data-test="cool-card-div"
  >
    hello I am a card :)
  </p>
   
  <p>
    Current Tokjbtal:
  </p>
   
  <p>
    567
  </p>
   
  <button>
    Increment
  </button>
</div>
`;
